### DISCUSSION TITLE
Property containing value from another property

### DISCUSSION THREAD

**Discussion by BaaderKristineKristensen** [wixtoolset/issues#7611] [Questions]
Hi, 

I have a settings file I want to check if exists in a specific folder before  I install the software. If it exists will I skip a if statement I have created. or else should it go inside the if. 

I read in a discussion that the property were set to the full path if a file in file-search exists. I then created a property containing the directory and file search setting. 

Then I wanted to create a expected path variable property to match with the result of the file search property. The second property does not show what I expected in my MS Install log ( which were generated by using msiexec command).

**How can I make sure the if statement look as the right expeted paths?**

The value of dir_Settings_Sharedis generated locally inside the wix using the Directory Id tag. 
The value of $(var.GlobalSettingsFile) is generated from powershell and sent through into the install file as a parameter. 

```
<Property Id='GLOBALSETTINGSFILESEARCH'>
     <DirectorySearch Id='Folder_dir_Settings_Shared' Path="[dir_Settings_Shared]">
        <FileSearch Name='$(var.GlobalSettingsFile)' />
     </DirectorySearch>
</Property>
<Property Id="GLOBALSETTINGSFILESEARCHALREADY" Value ="[dir_Settings_Shared].$(var.GlobalSettingsFile)"/>
```
...
```
<?if [GLOBALSETTINGSFILESEARCH]  !=  [GLOBALSETTINGSFILESEARCHALREADY]?>
	<!-- CODE TO GENERATE FILE-->
<?endif ?>

```
The outcome in logs are: 
Property(C): dir_Settings_Shared = C:\FolderName\setting\Shared\
...
AppSearch: Property: GLOBALSETTINGSFILESEARCH, Signature: Folder_dir_Settings_Shared
MSI (c) (E4:E4) [09:51:31:693]: Note: 1: 1322 2:  
MSI (c) (E4:E4) [09:51:31:693]: Note: 1: 1322 2:  
MSI (c) (E4:E4) [09:51:31:693]: Note: 1: 1325 2: GlobalSettings.xml 
...
Property(C): GLOBALSETTINGSFILESEARCHALREADY = [dir_Settings_Shared].GlobalSettings.xml
...

**Comment by robmen** [comment]
You're mixing up build time and install time variables. I did a series of Deployment Dojo episodes on exactly this topic, starting with [Episode 12](https://youtube.com/live/g1PBKhq2oe8). I bet if you watch through to Episode 18, it'll all make a lot more sense.

**Reply by BaaderKristineKristensen** [reply]
I'll try that, thank you so much :)

**Reply by BaaderKristineKristensen** [reply]
I just watched the episode, but still am having the same issue

**Reply by robmen** [reply]
>  I bet if you watch through to Episode 18, it'll all make a lot more sense.

I'm not sure Episode 12 alone will explain the concepts. It ended up being a 7-part video series to explain all the variable types available.

**Reply by BaaderKristineKristensen** [reply]
If you had a similar case how would you handle it?
